pkg/api/accessrule.go

```diff
@@ -189,6 +190,71 @@ func (a *API) AdminGetAccessRuleVersion(w http.ResponseWriter, r *http.Request,
 	apio.JSON(ctx, w, q.Result.ToAPIDetail(), http.StatusOK)
 }
 
+// Your GET endpoint
+// (GET /api/v1/access-rules/lookup)
+func (a *API) AccessRuleLookup(w http.ResponseWriter, r *http.Request, params types.AccessRuleLookupParams) {
+	ctx := r.Context()
+	// fetch all active access rules
+	u := auth.UserFromContext(ctx)
+	q := storage.ListAccessRulesForGroupsAndStatus{Groups: u.Groups, Status: rule.ACTIVE}
+	_, err := a.DB.Query(ctx, &q)
+	if err != nil && err != ddb.ErrNoItems {
+		apio.Error(ctx, w, err)
+		return
+	}
+
+	res := types.ListAccessRulesResponse{
+		AccessRules: []types.AccessRule{},
+	}
+
+	// Validate that params.Type is a valid provider id
+	// NOTE: we may want to validate provider versions, this should work for now
+	// Assumption: user is using the latest type
+	_, p, err := providerregistry.Registry().GetLatestByType(string(*params.Type))
+	if err != nil {
+		apio.Error(ctx, w, err)
+	}
+
+	//	filter by params.AccountId
+	for _, r := range q.Result {
+		switch p.DefaultID {
+		case "aws-sso-v2":
+			// we must support string and []string for With/WithSelectable
+			ruleAccIds := []string{}
+			singleRuleAccId, found := r.Target.ToAPI().With.Get("accountId")
+			if !found {
+				ruleAccIds, found = r.Target.ToAPI().WithSelectable.Get("accountId")
+				if !found {
+					continue // if not found continue
+				}
+			} else {
+				ruleAccIds = append(ruleAccIds, singleRuleAccId)
+			}
+
+			for _, ruleAccId := range ruleAccIds {
+				reqAccId := (*params.AccountId)
+				if reqAccId != ruleAccId {
+					continue // if not found continue
+				}
+				// lookup ProviderOptions for given rule and get the
+				q := storage.GetProviderOptions{ProviderID: p.DefaultID}
+				_, err := a.DB.Query(ctx, &q)
+				if err != nil {
+					continue // todo: log error
```

Comment:
This is what you're looking for:

```suggestion
					logger.Get(ctx).Errorw("error finding provider options", zap.Error(err))
					continue
```